04_rag_tuning/rag.py

```diff
@@ -0,0 +1,48 @@
+import faiss
+import lamini
+import jsonlines
+
+EMBEDDING_SIZE = 384
+k = 2
+
+# Set up for the index, which holds the embeddings, and the splits, which holds the corresponding plain text
+index = faiss.IndexFlatL2(EMBEDDING_SIZE)
+splits = []
```

Comment:
Why is EMBEDDING_SIZE hardcoded and can it be changed..?

Reply:
our default embedding model is dimension 384

Reply:
This embedding size needs to match the embedding model that's used. Previously the way we auto-detected this was by sending an embedding request first and then getting the embedding size from that. We could go back to doing that if we want to! Code is in the old lamini_index

Reply:
It looks like we hardcoded it before as well: https://github.com/lamini-ai/lamini-examples/blob/a3b747795996d4a70fedc730a62c0480ab17c6a3/04_rag_tuning/lamini_rag/lamini_index.py#L39

Reply:
I can look into a way to detect it, let me know if you have any ideas. I don't think users should be expected to know the embedding size the model is using

Reply:
Fixed 👍